hail/src/main/scala/is/hail/expr/ir/functions/ArrayFunctions.scala

```diff
@@ -304,7 +304,9 @@ object ArrayFunctions extends RegistryFunctions {
       ToArray(StreamFlatMap(ToStream(a), elt.name, ToStream(elt)))
     }
 
-    registerCodeWithMissingness("corr", TArray(TFloat64), TArray(TFloat64), TFloat64, null) {
+    registerCodeWithMissingness("corr", TArray(TFloat64), TArray(TFloat64), TFloat64, {
+      (t1: PType, t2: PType) => InferPType.getNestedElementPTypes(Seq(t1, t2)).asInstanceOf[PArray].elementType.orMissing(t1.required & t2.required)
```

Comment:
There's a missing case even when both sides are defined -- this has to return `PFloat64(false))`.